libvast/src/system/partition.cpp

```diff
@@ -576,17 +581,15 @@ passive_partition(caf::stateful_actor<passive_partition_state>* self, uuid id,
         return self->quit();
       }
       // Deserialize chunk from the filesystem actor
-      auto view
-        = span(reinterpret_cast<const byte*>(chunk->data()), chunk->size());
-      auto partition
-        = fbs::as_versioned_flatbuffer<fbs::Partition>(view, fbs::Version::v1);
-      if (!partition) {
-        VAST_ERROR(self, "failed to parse provided chunk as flatbuffer:",
-                   render(partition.error()));
-        return self->quit(std::move(partition.error()));
+      auto partition = fbs::GetPartition(chunk->data());
+      if (partition->versioned_partition_type()
+          != fbs::PartitionUnion::v1_Partition) {
+        VAST_ERROR(self, "got flatbuffer partition with invalid version");
+        return self->quit();
       }
+      auto partition_v1 = partition->versioned_partition_as_v1_Partition();
       self->state.partition_chunk = chunk;
-      self->state.flatbuffer = *partition;
+      self->state.flatbuffer = partition_v1;
```

Comment:
This can just be merged with the line above, no need for the additional variable.